src/App.js

```diff
@@ -1,16 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+  const [likeCount, setLikes] = useState(0);
+  // const contacts = [];
+
+  // for (const entry of entries) {
+  //   if (!contacts.includes(entry.sender)) {
+  //     contacts.push(entry.sender);
+  //   }
+  // }
+
+  const handleLikes = (id) => {
+    const newEntries = [];
+    for (const entry of entries) {
+      const newEntry = { ...entry };
+      if (newEntry.id === id) {
+        newEntry.liked = !newEntry.liked;
+        likeCounter(newEntry.liked);
+      }
+      newEntries.push(newEntry);
+    }
+    setEntries(newEntries);
+  };
+
+  const likeCounter = (liked) => {
+    if (liked === true) {
+      setLikes(likeCount + 1);
+    } else if (liked === false) {
+      setLikes(likeCount - 1);
+    }
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat Log Between {entries[0].sender} and {entries[1].sender}
```

Comment:
Tip: This assumes that it's a chat with only two people (which is fine) but it also assumes that the chats start alternating, which might not be true in every case.

This would be more robust if it looked for the first sender that wasn't the same as `entries[0].sender` though that would likely require writing a helper function so might not be worth it.  

Just thought it was an approach you might find interesting.  🙂